django/db/migrations/autodetector.py

```diff
@@ -1694,6 +1697,27 @@ def _get_dependencies_for_generated_field(self, field):
                 )
         return dependencies
 
+    def _get_generated_field_dependencies_for_removed_field(
+        self, app_label, model_name, field_name
+    ):
+        dependencies = []
+        model_state = self.from_state.models[app_label, model_name]
+        generated_fields = (f for f in model_state.fields.values() if f.generated)
+        for field in generated_fields:
+            if any(
+                field_name == name
+                for name, *_ in models.Model._get_expr_references(field.expression)
+            ):
+                dependencies.append(
+                    OperationDependency(
+                        app_label,
+                        model_name,
+                        field.name,
+                        OperationDependency.Type.REMOVE,
+                    )
+                )
```

Comment:
```suggestion
            dependencies.append(
                OperationDependency(
                    app_label,
                    model_name,
                    field.name,
                    OperationDependency.Type.REMOVE,
                )
            )
```
I can update to the following without test failures

Reply:
I get the following:
<details>
<summary>Errors</summary>
<code>
ERROR: test_remove_generated_field_and_one_of_multiple_base_fields (migrations.test_autodetector.AutodetectorTests.test_remove_generated_field_and_one_of_multiple_base_fields)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 651, in run
    self._callTestMethod(testMethod)
    
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 606, in _callTestMethod
    if method() is not None:
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/tests/migrations/test_autodetector.py", line 1540, in test_remove_generated_field_and_one_of_multiple_base_fields
    changes = self.get_changes(initial_state, updated_state)
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/tests/migrations/test_autodetector.py", line 222, in get_changes
    )._detect_changes()
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/django/db/migrations/autodetector.py", line 226, in _detect_changes
    self._sort_migrations()
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/django/db/migrations/autodetector.py", line 432, in _sort_migrations
    self.generated_operations[app_label] = list(ts.static_order())
    ^^^^^^^^^^^
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/graphlib.py", line 244, in static_order
    self.prepare()
    ~~~~~~~~~~~~^^
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/graphlib.py", line 106, in prepare
    raise CycleError(f"nodes are in a cycle", cycle)
    ^^^^^^^^^^^
graphlib.CycleError: ('nodes are in a cycle', [<RemoveField  model_name='author', name='full_name'>, <RemoveField  model_name='author', name='full_name'>])

======================================================================
ERROR: test_remove_generated_field_before_its_base_field (migrations.test_autodetector.AutodetectorTests.test_remove_generated_field_before_its_base_field)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 651, in run
    self._callTestMethod(testMethod)
    
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 606, in _callTestMethod
    if method() is not None:
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/tests/migrations/test_autodetector.py", line 1476, in test_remove_generated_field_before_its_base_field
    changes = self.get_changes(initial_state, updated_state)
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/tests/migrations/test_autodetector.py", line 222, in get_changes
    )._detect_changes()
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/django/db/migrations/autodetector.py", line 226, in _detect_changes
    self._sort_migrations()
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/django/db/migrations/autodetector.py", line 432, in _sort_migrations
    self.generated_operations[app_label] = list(ts.static_order())
    ^^^^^^^^^^^
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/graphlib.py", line 244, in static_order
    self.prepare()
    ~~~~~~~~~~~~^^
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/graphlib.py", line 106, in prepare
    raise CycleError(f"nodes are in a cycle", cycle)
    ^^^^^^^^^^^
graphlib.CycleError: ('nodes are in a cycle', [<RemoveField  model_name='author', name='upper_name'>, <RemoveField  model_name='author', name='upper_name'>])

======================================================================
ERROR: test_remove_generated_field_before_multiple_base_fields (migrations.test_autodetector.AutodetectorTests.test_remove_generated_field_before_multiple_base_fields)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 651, in run
    self._callTestMethod(testMethod)
    
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/unittest/case.py", line 606, in _callTestMethod
    if method() is not None:
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/tests/migrations/test_autodetector.py", line 1502, in test_remove_generated_field_before_multiple_base_fields
    changes = self.get_changes(initial_state, updated_state)
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/tests/migrations/test_autodetector.py", line 222, in get_changes
    )._detect_changes()
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/django/db/migrations/autodetector.py", line 226, in _detect_changes
    self._sort_migrations()
    ^^^^^^^^^^^^^^^
  File "/home/clifford/code/django/django/db/migrations/autodetector.py", line 432, in _sort_migrations
    self.generated_operations[app_label] = list(ts.static_order())
    ^^^^^^^^^^^
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/graphlib.py", line 244, in static_order
    self.prepare()
    ~~~~~~~~~~~~^^
  File "/home/clifford/.pyenv/versions/3.13.2/lib/python3.13/graphlib.py", line 106, in prepare
    raise CycleError(f"nodes are in a cycle", cycle)
    ^^^^^^^^^^^
graphlib.CycleError: ('nodes are in a cycle', [<RemoveField  model_name='author', name='full_name'>, <RemoveField  model_name='author', name='full_name'>])

----------------------------------------------------------------------
</code>
</details>